src/sign_workflow/sign_artifacts.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+from abc import abstractmethod
+from pathlib import Path
+
+from manifests.build_manifest import BuildManifest
+from sign_workflow.signer import Signer
+
+
+class SignArtifacts:
+    def __init__(self, target: Path, component, artifact_type, signature_type):
+        self.target = target
+        self.component = component
+        self.artifact_type = artifact_type
+        self.signature_type = signature_type
+        self.signer = Signer()
+
+    @abstractmethod
+    def __sign__(self):
+        pass
+
+    def sign(self):
+        self.__sign__()
+        logging.info("Done.")
+
+    def __sign_artifacts__(self, artifacts, basepath, signature_type):
```

Comment:
The `signature_type` argument here is redundant because it's already passed into the constructor, so you can do `self.signer.sign_artifacts(artifacts, basepath, self.signature_type)`. Remove it.